### DISCUSSION TITLE
建议增加自定义API模型名功能

### DISCUSSION THREAD

**Discussion by youjiaping123** [ZuodaoTech/everyone-can-use-english#662] [Ideas]
首先感谢您开发了这款方便好用的软件

我想提一个小建议，不知是否可以考虑在软件中增加自定义模型名称的功能。因为除了 OpenAI，我有时也需要使用其他厂商提供的 API，比如 DeepSeek API。DeepSeek 提供了一些 OpenAI 没有的模型，如 deepseek-chat 和 deepseek-coder，但目前软件只允许选择 OpenAI 的模型名称，无法使用这些模型。

如果能够在软件中增加一个自定义模型名称的输入框，让用户可以手动输入任意的模型名称，那就可以兼容包括 DeepSeek 在内的更多 API 提供商，让软件的适用范围更广。即使将来出现了新的 API 和模型，也无需频繁更新软件，用户只需要手动输入相应的模型名称即可。

我想这个功能对您来说应该不难实现，但会给用户带来很大的方便。如果您能考虑增加这个功能，我和其他用户将不胜感激。

**Comment by an-lee** [comment]
不是很明白你的意思。你是有 deepseek 的 api key，想要 Enjoy 也支持 deepseek 设置和调用？还是说，你没有 deepseek 的 api key，只是想要用 deepseek 的模型？

如果是后一种，你在 Enjoy 里把 AI 引擎改为 EnjoyAI，除了 OpenAI 的模型，也有其他的模型供选择，这个列表会持续更新。deepseek 的模型也可以加上。

**Reply by an-lee** [reply]
![image](https://github.com/xiaolai/everyone-can-use-english/assets/22209056/bc72fa6d-f336-4623-a7a5-afbc33e12926)

**Comment by youjiaping123** [comment]
是前一种，因为deepseek兼容openai的api所以只需要增加一个自定义模型名的功能将模型名手动输入选择为deepseek的模型也就可以直接使用了。
![1717900804010](https://github.com/xiaolai/everyone-can-use-english/assets/109498374/68c93a70-3b13-47b3-82d6-36350ac61e2c)

**Reply by an-lee** [reply]
嗯，可以支持一下。

**Reply by an-lee** [reply]
#876 支持了这个功能。
